### ISSUE TITLE
Auth keys are invalid after server restart

### ISSUE THREAD

**Issue by OrganicCat** [j256/two-factor-auth#31]
The keys work fine for any period of time after generating, as long as the server continues to run.

Is there some link between validation and the server that gets stored in cache, and wiped upon restart? Not doing anything custom, just generating QR codes and keys, scanning them with Google Authenticator, and using the validation engine built in. It works 100% of the time right up until a server restart. The secret is stored securely, regardless of what's going on, and it doesn't change when rebooting or anything like that.

Is there a step I'm missing that's causing this issue?

I know the great solution would be never dropping the server, haha, but we can't have our users set up auth again every time the server needs to be updated or rebooted.

Reproducing:

1) Generate new secret
2) Add secret to google auth with QR code scanner
3) Test google key with auth system 👍 
4) Restart server
5) Test google key with auth system again (failure 👎 )

Suggestion to the error message as well:

"Unable to invoke Cipher due to bad padding" might make some sense for the developer of the library, but it gives no indication to an end user of the library that the key they're using is invalid. Nobody I've worked with had any clue what it meant, and it was only after troubleshooting another issue that we came to see it as "The auth key doesn't match the secret"

**Comment by OrganicCat** [comment]
Was incorrectly using a random number in a bean that was randomized on startup (and never stored anywhere). The salt and passphrase, etc. must be stored (securely) and not random for the encryption to work after being generated again.
